scripts/world_bank/worldbank_ids/download.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.licenses/org/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import logging
+import datetime
+# Import the required function after installing the package
+from bblocks import WorldBankData, DebtIDS
+from time import time
+
+logging.basicConfig(level=logging.INFO)
+start_time = time()
+DEFAULT_INPUT_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                  "input_files")
+
+if not os.path.exists(DEFAULT_INPUT_PATH):
+    logging.info(f"Creating folder -> {DEFAULT_INPUT_PATH}")
+    os.mkdir(DEFAULT_INPUT_PATH)
+
+# create a list of indicators under each dataset.
+
+interest = [
+    "DT.INT.BLAT.CD", "DT.INT.BLTC.CD", "DT.INT.BLTC.CD", "DT.INT.DLXF.CD",
+    "DT.INT.DPNG.CD", "DT.INT.DPPG.CD", "DT.INT.MIBR.CD", "DT.INT.MIDA.CD",
+    "DT.INT.MLAT.CD", "DT.INT.MLTC.CD", "DT.INT.OFFT.CD", "DT.INT.PBND.CD",
+    "DT.INT.PCBK.CD", "DT.INT.PNGB.CD", "DT.INT.PNGC.CD", "DT.INT.PROP.CD",
+    "DT.INT.PRVT.CD"
+]
+
+principal = [
+    "DT.AMT.DLXF.CD", "DT.AMT.DPPG.CD", "DT.AMT.DPPG.CD", "DT.AMT.OFFT.CD",
+    "DT.AMT.MLAT.CD", "DT.AMT.MIBR.CD", "DT.AMT.MIDA.CD", "DT.AMT.MLTC.CD",
+    "DT.AMT.BLAT.CD", "DT.AMT.BLTC.CD", "DT.AMT.PRVT.CD", "DT.AMT.PBND.CD",
+    "DT.AMT.PCBK.CD", "DT.AMT.PROP.CD", "DT.AMT.DPNG.CD", "DT.AMT.PNGB.CD",
+    "DT.AMT.PNGC.CD"
+]
+
+disbursed = [
+    "DT.DOD.DLXF.CD", "DT.DOD.DPPG.CD", "DT.DOD.DPPG.CD", "DT.DOD.OFFT.CD",
+    "DT.DOD.MLAT.CD", "DT.DOD.MIBR.CD", "DT.DOD.MIDA.CD", "DT.DOD.MLTC.CD",
+    "DT.DOD.BLAT.CD", "DT.DOD.BLTC.CD", "DT.DOD.PRVT.CD", "DT.DOD.PBND.CD",
+    "DT.DOD.PCBK.CD", "DT.DOD.PROP.CD", "DT.DOD.DPNG.CD", "DT.DOD.PNGB.CD",
+    "DT.DOD.PNGC.CD"
+]
+
+currency = [
+    "DT.CUR.DMAK.ZS", "DT.CUR.EURO.ZS", "DT.CUR.EURO.ZS", "DT.CUR.JYEN.ZS",
+    "DT.CUR.USDL.ZS", "DT.CUR.SDRW.ZS", "DT.CUR.OTHC.ZS", "DT.CUR.UKPS.ZS",
+    "DT.CUR.SWFR.ZS", "DT.CUR.MULC.ZS", "DT.CUR.FFRC.ZS"
+]
+
+indicator_list = [interest, currency, principal, disbursed]
+indicator_listname = ["interest", "currency", "principal", "disbursed"]
+start_year = 1970
+end_year = datetime.date.today().year + 5
+
+# Itterating each list to download the respective data.
+for idx, indicator in enumerate(indicator_list):
```

Comment:
1. Add try catch block and log error if any.
2. Log once each indicator data load success to get to know the status.

Reply:
The bblocks package has this inbuilt where if the indicator does not exist, it will throw an error.